File: Pod/Classes/Core/ARComponentViewController.m
#import "ARComponentViewController.h"
#import "AREmission.h"

#import <React/RCTRootView.h>

@interface ARComponentViewController ()
@property (nonatomic, strong, readonly) AREmission *emission;
@property (nonatomic, strong, readonly) NSString *moduleName;
@end

@implementation ARComponentViewController

- (instancetype)initWithModuleName:(nonnull NSString *)moduleName;
{
  return [self initWithEmission:[AREmission sharedInstance] moduleName:moduleName];
}

- (instancetype)initWithEmission:(AREmission *)emission moduleName:(NSString *)moduleName;
{
  if ((self = [super initWithNibName:nil bundle:nil])) {
    _emission = emission;
    _moduleName = moduleName;
  }
  return self;
}

- (void)viewDidLoad;
{
  [super viewDidLoad];
  
  RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:self.emission.bridge
                                                   moduleName:self.moduleName
                                            initialProperties:nil];
  [self.view addSubview:rootView];
  rootView.reactViewController = self;
  
  rootView.translatesAutoresizingMaskIntoConstraints = NO;
  [self.view addConstraints:@[
    [NSLayoutConstraint constraintWithItem:rootView
                                 attribute:NSLayoutAttributeTop
                                 relatedBy:NSLayoutRelationEqual
                                    toItem:self.topLayoutGuide
                                 attribute:NSLayoutAttributeBottom
                                multiplier:1
                                  constant:0],
    [NSLayoutConstraint constraintWithItem:rootView
                                 attribute:NSLayoutAttributeLeading
                                 relatedBy:NSLayoutRelationEqual
                                    toItem:self.view
                                 attribute:NSLayoutAttributeLeading
                                multiplier:1
                                  constant:0],
    [NSLayoutConstraint constraintWithItem:rootView
                                 attribute:NSLayoutAttributeTrailing
                                 relatedBy:NSLayoutRelationEqual
                                    toItem:self.view
                                 attribute:NSLayoutAttributeTrailing
                                multiplier:1
                                  constant:0],
    [NSLayoutConstraint constraintWithItem:rootView
                                 attribute:NSLayoutAttributeBottom
                                 relatedBy:NSLayoutRelationEqual
                                    toItem:self.bottomLayoutGuide
                                 attribute:NSLayoutAttributeTop
                                multiplier:1
                                  constant:0]
  ]];
}

@end

File: Pod/Classes/ARComponentViewController.swift
import UIKit
import React

public class ARComponentViewController : UIViewController {
  private var bridge: RCTBridge!
  private var moduleName: String!
  
  public init(bridge: RCTBridge!, moduleName: String!) {
    self.bridge = bridge
    self.moduleName = moduleName
    super.init(nibName: nil, bundle: nil)
  }

  required public init?(coder aDecoder: NSCoder) {
      fatalError("init(coder:) oh, Swift")
  }

  override public func viewDidLoad() {
    super.viewDidLoad()

    let rootView = RCTRootView(bridge: bridge, moduleName: moduleName, initialProperties: nil)
    view.addSubview(rootView)

    rootView.translatesAutoresizingMaskIntoConstraints = false
    view.addConstraints([
      NSLayoutConstraint.init(item: rootView, attribute: .Top, relatedBy: .Equal, toItem: topLayoutGuide, attribute: .Bottom, multiplier: 1, constant: 0),
      NSLayoutConstraint.init(item: rootView, attribute: .Leading, relatedBy: .Equal, toItem: view, attribute: .Leading, multiplier: 1, constant: 0),
      NSLayoutConstraint.init(item: rootView, attribute: .Trailing, relatedBy: .Equal, toItem: view, attribute: .Trailing, multiplier: 1, constant: 0),
      NSLayoutConstraint.init(item: rootView, attribute: .Bottom, relatedBy: .Equal, toItem: bottomLayoutGuide, attribute: .Top, multiplier: 1, constant: 0),
    ])
  }
}